context/src/test/java/io/grpc/ContextTest.java

```diff
@@ -745,4 +747,53 @@ public void expiredDeadlineShouldCancelContextImmediately() {
     assertTrue(context.isCancelled());
     assertThat(context.cancellationCause(), instanceOf(TimeoutException.class));
   }
+
+  /**
+   * Tests initializing the {@link Context} class with a custom logger which uses Context's storage
+   * when logging.
+   */
+  @Test
+  public void initContextWithCustomClassLoaderWithCustomLogger() throws Exception {
+    StaticTestingClassLoader classLoader =
+        new StaticTestingClassLoader(
+            getClass().getClassLoader(),
+            Pattern.compile("(io\\.grpc\\.Context.*)|(io\\.grpc\\.ThreadLocalContextStorage.*)"));
+    Class<?> runnable =
+        classLoader.loadClass("io.grpc.ContextTest$LoadMeWithStaticTestingClassLoader");
+
+    ((Runnable) runnable.getDeclaredConstructor().newInstance()).run();
+  }
+
+  // UsedReflectively
+  @VisibleForTesting
```

Comment:
VisibleForTesting is a strange annotation here. I'd get rid of it and just have the UsedReflectively.